### PR TITLE
teacher adpator freeze + map location cpu for all pretrained models

### PR DESCRIPTION
**Pull request by shaydeci**
Imagenet resent50 KD recipe was broken because of changes in https://github.com/Deci-AI/super-gradients/pull/319 .
(Input adaptor's parameters had require_grad=True which led to CUDA OOM).
Additionally map location=None in load_pretrained_weights caused to having extra GPU memory allocated in rank:0 when training in DDP. I removed it since the net is moved to the proper device in Trainer anyways.

### CODE REVIEW

**Review by yurkovak — APPROVED** [review]
LGTM
